forms-flow-api/tests/unit/api/test_form_process_mapper.py

```diff
@@ -854,3 +855,25 @@ def test_form_list_submission_count(app, client, session, jwt, create_mapper):
     forms = response.json["forms"]
     assert len(forms) == 1
     assert "submissionsCount" not in forms[0]
+
+
+def test_get_form_data(app, client, session, jwt, mock_redis_client, create_mapper):
+    """Testing get form data endpoint."""
+    token = get_token(jwt, role=CREATE_DESIGNS, username="designer")
+    headers = {"Authorization": f"Bearer {token}", "content-type": "application/json"}
+    payload = get_formio_form_request_payload()
+    payload["componentChanged"] = True
+    payload["newVersion"] = True
+    response = client.post("/form/form-design", headers=headers, json=payload)
+    assert response.status_code == 201
+    form_id = response.json["_id"]
+    response = client.get(f"/form/form-data/{form_id}?authType=designer", headers=headers)
+    assert response.status_code == 200
+    assert response.json is not None
+    unauthorized_token = get_token(jwt, role=CREATE_DESIGNS, username="designer2")
+    headers = {
+        "Authorization": f"Bearer {unauthorized_token}",
+        "content-type": "application/json",
+    }
+    response = client.get(f"/form/form-data/{form_id}?authType=designer", headers=headers)
+    assert response.status_code == 403
```

Comment:
can you please add test cases for other authTypes too?

Reply:
sure